### PR TITLE
deps: Bump OpenTelemetry-Go Contrib 0.45.0

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
This version includes security fixes which addresses vulnerability CVE-2023-45142.More details can be
found at https://github.com/open-telemetry/opentelemetry-go-contrib/security/advisories/GHSA-rcjv-mgp8-qvmr.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -4834,30 +4830,10 @@ func TestDistributedTracingEnabled(t *testing.T) {
 		}}`)
 
 	ctx := context.Background()
-	_, tracerProvider, err := distributedtracing.Init(ctx, c, "foo")
+	_, _, err := distributedtracing.Init(ctx, c, "foo")
```
Comment (ashutosh-narkar) [inline server/server_test.go:4833]: `otelhttp.Handler` type was removed in https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2964. See [this](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/2964#discussion_r1177591249) comment for details.

**Review by johanfylling — COMMENTED** [review]
The version bump looks ok to me 👍. 
I'm a bit uneasy about the altered/removed tests, though. That said. I'm not familiar with this library, so this might still be fine. Would like someone else to also weigh in.

**Comment by srenatus** [comment]
I've also been trying to find if we're using the replacement for the Handler and couldn't find it. 🤔 
But also, we've got a whole bunch of e2e tests here: https://github.com/open-policy-agent/opa/blob/main/test/e2e/distributedtracing/distributedtracing_test.go -- if they pass, we should be 🆗

**Comment by ashutosh-narkar** [comment]
> I've also been trying to find if we're using the replacement for the Handler and couldn't find it. 🤔 But also, we've got a whole bunch of e2e tests here: https://github.com/open-policy-agent/opa/blob/main/test/e2e/distributedtracing/distributedtracing_test.go -- if they pass, we should be 🆗

@srenatus @johanfylling do y'all think we can get this in then?

**Review by johanfylling — APPROVED** [review]
Tests are passing, so I'll green-light this 👍 .
We should look extend the testing around the `WithDistributedTracingOpts()` func, though, as that's part of the public SDK. But we can do that as a separate task.

**Pull request merged**